File: frontend/js/semantic-icons.js
(function () {
  "use strict";

  const names = Object.freeze({
    "action.add": "plus",
    "action.close": "x",
    "action.more": "ellipsis",
    "action.refresh": "refresh-cw",
    "action.search": "search",
    "action.send": "send",
    "action.attach": "paperclip",
    "action.copy": "copy",
    "action.edit": "pencil",
    "action.delete": "trash-2",
    "action.play": "play",
    "action.stop": "square",
    "action.resume": "rotate-ccw",
    "action.settings": "settings",
    "action.terminal": "terminal",
    "navigation.expand": "chevron-down",
    "navigation.collapse": "chevron-up",
    "navigation.back": "arrow-left",
    "navigation.forward": "arrow-right",
    "entity.chat": "message-square",
    "entity.project": "folder",
    "entity.workflow": "workflow",
    "entity.mobile": "smartphone",
    "state.info": "info",
    "state.success": "circle-check",
    "state.warning": "triangle-alert",
    "state.danger": "circle-x",
    "state.unknown": "circle-help",
    "state.loading": "loader-circle"
  });

  function iconUrl(semanticId) {
    const name = names[semanticId];
    return name ? `icons/lucide/${name}.svg` : "";
  }

  function hydrate(root = document) {
    root.querySelectorAll("[data-semantic-icon]").forEach((control) => {
      if (control.querySelector(":scope > .semantic-icon")) return;
      const url = iconUrl(control.dataset.semanticIcon);
      if (!url) return;
      const icon = document.createElement("span");
      icon.className = "semantic-icon";
      icon.setAttribute("aria-hidden", "true");
      icon.style.setProperty("--semantic-icon-url", `url(\"${url}\")`);
      control.prepend(icon);
    });
  }

  window.SPIRITKIN_ICONS = Object.freeze({ names, iconUrl, hydrate });
  if (document.readyState === "loading") {
    document.addEventListener("DOMContentLoaded", () => hydrate(), { once: true });
  } else {
    hydrate();
  }
})();

File: ios/SpiritKinTerminal/Sources/Support/SemanticIcons.swift
import Foundation

enum SemanticIcons {
    static let names: [String: String] = [
        "action.add": "plus", "action.close": "xmark", "action.more": "ellipsis",
        "action.refresh": "arrow.clockwise", "action.search": "magnifyingglass",
        "action.send": "paperplane", "action.attach": "paperclip", "action.copy": "doc.on.doc",
        "action.edit": "pencil", "action.delete": "trash", "action.play": "play.fill",
        "action.stop": "stop.fill", "action.resume": "arrow.counterclockwise",
        "action.settings": "gearshape", "action.terminal": "apple.terminal",
        "navigation.expand": "chevron.down", "navigation.collapse": "chevron.up",
        "navigation.back": "chevron.backward", "navigation.forward": "chevron.forward",
        "entity.chat": "message", "entity.project": "folder",
        "entity.workflow": "point.3.connected.trianglepath.dotted", "entity.mobile": "iphone",
        "state.info": "info.circle", "state.success": "checkmark.circle",
        "state.warning": "exclamationmark.triangle", "state.danger": "xmark.circle",
        "state.unknown": "questionmark.circle", "state.loading": "progress.indicator"
    ]

    static func systemName(_ semanticID: String) -> String {
        names[semanticID] ?? names["state.unknown"]!
    }
}
